src/core/mac/mac_types.cpp

```diff
@@ -155,6 +155,7 @@ Error NetworkName::Set(const char *aNameString)
     Error    error;
     NameData data(aNameString, kMaxSize + 1);
 
+    VerifyOrExit(data.GetLength() >= 1, error = kErrorInvalidArgs);
```

Comment:
I just notcied this. This check here does nothing and doe not really check the length of the new name.
- Note that `NameData` is a warpper over a pointer to some char buffer and a length. 
- From `NameData data(aNameString, kMaxSize + 1)` we init the `data` with length `kMaxSize + 1`.
- So `data.GetLength()` will be `kMaxSize + 1` which is always `>=1`.... 😃 

I will submit a PR to update and fix this.